docs/sdks/server-sdks/python.md

```diff
@@ -322,3 +322,130 @@ if self.campaign_json is not None:
+            categorical_attributes={"image_aspect_ratio": "16:9"},
+        ),
+    ],
+    "control",
+)
+```
+
+#### Subject Context
+
+The subject context contains contextual information about the subject that is independent of bandit actions.
+For example, the subject's age or country.
+
+The subject context has type `Attributes` which has two fields:
+
+- `numeric_attributes` (Dict[str, float]): A dictionary of numeric attributes (such as "age")
+- `categorical_attributes` (Dict[str, str]): A dictionary of categorical attributes (such as "country")
+
+:::note
+The `categerical_attributes` are also used for targeting rules for the feature flag similar to how `subject_attributes` are used for that with regular feature flags. 
+:::
+
+#### Action Contexts
+
+Next, supply a list with actions and their contexts: `actions_with_contexts: List[ActionContext]`.
+If the user is assigned to the bandit, the bandit selects one of the actions supplied here,
+and all actions supplied are considered to be valid; if an action should not be shown to a user, it should be filtered out of this list.
+
+`ActionContexts` contain the action key, as well as the `Attributes` for the action. This is similar to the `subject_context`, but it is for the action.
+They are conveniently constructed using the `ActionContext.create` method:
+```python
+eppo_client.bandit.ActionContext.create(
+    "puma",
+    numeric_attributes={"brand_affinity": 2.3},
+    categorical_attributes={"image_aspect_ratio": "16:9"},
+)
+```
+
+Note that action contexts can contain two kinds of information:
+- Action specific context: e.g. the image aspect ratio of image corresponding to this action
+- User-action interaction context: e.g. there could be a "brand-affinity" model that computes brand affinties of users to brands, and scores of this model can be added to the action context to provide additional context for the bandit.
+
+#### Result
+
+The `bandit_result` is an instance of `BanditResult`, which has two fields:
+
+- `variation` (str): The variation that was assigned to the subject
+- `action` (Optional[str]): The action that was assigned to the subject
+
+The variation returns the feature flag variation, this can be the bandit itself, or the "status quo" variation if the user is not assigned to the bandit.
+If we are unable to generate a variation, for example when the flag is turned off, then the `default` variation is returned. 
+In both of those cases, the `action` is `None`, and you should use the status-quo algorithm to select an action.
+
+When `action` is not `None`, the bandit has selected that action to be shown to the user.
+
+#### Status quo algorithm
+
+In order to accurately measure the performance of the bandit, we need to compare it to the status quo algorithm using an experiment.
+This status quo algorithm could be a complicated algorithm to that selects an action according to a different model, or a simple baseline such as selecting a fixed or random action.
```

Comment:
Should we explicitly mention this is what is done when the non-bandit variation / null action is selected?

Reply:
Added some more detail